File: 3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i.cpp
int init=[]{ios_base::sync_with_stdio(false);cin.tie(0);cout.tie(0);return 0;}();
class Solution {
public:
    int minFlips(vector<vector<int>>& grid) {
        int m = grid.size() / 2, n = grid[0].size() / 2;
        int row_ans = 0, col_ans = 0;

        for (const auto& row : grid) {
            for (int i = 0; i < n; ++i) {
                row_ans += row[i] ^ row[row.size() - i - 1];
            }
        }

        for (int j = 0; j < grid[0].size(); ++j) {
            for (int i = 0; i < m; ++i) {
                col_ans += grid[i][j] ^ grid[grid.size() - i - 1][j];
            }
        }

        return min(row_ans, col_ans);
    }
};

File: 3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i.rs
impl Solution {
    pub fn min_flips(grid: Vec<Vec<i32>>) -> i32 {
        let m = grid.len() / 2;
        let n = grid[0].len() / 2;
        let mut row_ans = 0;
        let mut col_ans = 0;

        for row in &grid {
            row_ans += (0..n).map(|i| row[i] ^ row[row.len() - i - 1]).sum::<i32>();
        }

        for j in 0..grid[0].len() {
            for i in 0..m {
                col_ans += grid[i][j] ^ grid[grid.len() - i - 1][j];
            }
        }

        row_ans.min(col_ans)
    }
}